test/Api/HookTest.php

```diff
@@ -0,0 +1,222 @@
+<?php
+
+namespace DevCycle\Test\Api;
+
+use DevCycle\HTTPConfiguration;
+use DevCycle\Model\DevCycleOptions;
+use DevCycle\Api\DevCycleClient;
+use DevCycle\Model\DevCycleUser;
+use DevCycle\Model\EvalHook;
+use DevCycle\Model\HookContext;
+use DevCycle\Model\BeforeHookError;
+use DevCycle\Model\AfterHookError;
+use PHPUnit\Framework\TestCase;
+
+/**
+ * HookTest Class
+ *
+ * Tests for evaluation hook functionality
+ *
+ * @category Class
+ * @package  DevCycle
+ */
+final class HookTest extends TestCase
+{
+    private DevCycleClient $client;
+    private DevCycleUser $user;
+
+    /**
+     * Setup before running each test case
+     */
+    public function setUp(): void
+    {
+        $this->user = new DevCycleUser(array(
+            "user_id" => "test-hook-user"
+        ));
+    }
+
+    /**
+     * Test that hooks work correctly with before, after, onFinally, and error callbacks
+     */
+    public function testHooksWorkCorrectly()
+    {
+        $beforeCalled = false;
+        $afterCalled = false;
+        $onFinallyCalled = false;
+        $errorCalled = false;
+
+        $hook = new EvalHook(
+            before: function (HookContext $context) use (&$beforeCalled) {
+                $beforeCalled = true;
+                $this->assertEquals('test-key', $context->getKey());
+                $this->assertEquals('test-default', $context->getDefaultValue());
+                $this->assertSame($this->user, $context->getUser());
+                $this->assertNull($context->getVariableDetails());
+            },
+            after: function (HookContext $context) use (&$afterCalled) {
+                $afterCalled = true;
+                $this->assertEquals('test-key', $context->getKey());
+                $this->assertEquals('test-default', $context->getDefaultValue());
+                $this->assertSame($this->user, $context->getUser());
+                $this->assertNotNull($context->getVariableDetails());
+            },
+            onFinally: function (HookContext $context) use (&$onFinallyCalled) {
+                $onFinallyCalled = true;
+                $this->assertEquals('test-key', $context->getKey());
+                $this->assertEquals('test-default', $context->getDefaultValue());
+                $this->assertSame($this->user, $context->getUser());
+            },
+            error: function (HookContext $context, \Exception $error) use (&$errorCalled) {
+                $errorCalled = true;
+                $this->assertEquals('test-key', $context->getKey());
+                $this->assertEquals('test-default', $context->getDefaultValue());
+                $this->assertSame($this->user, $context->getUser());
+            }
+        );
+
+        $options = new DevCycleOptions(false, null, null, [], [$hook]);
+        $this->client = new DevCycleClient(
+            sdkKey: getenv("dvc_server_token_hash") ?: "dvc_server_test_key",
```

Comment:
this is getting the env variable - you had the right env var but didn't update the default
```suggestion
            sdkKey: getenv("DEVCYCLE_SERVER_SDK_KEY") ?: "dvc_server_token_hash",
```